pkgs/tools/security/afl/qemu.nix

```diff
@@ -37,14 +37,15 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  patches =
-    [ ./qemu-patches/elfload.patch
-      ./qemu-patches/cpu-exec.patch
-      ./qemu-patches/no-etc-install.patch
-      ./qemu-patches/translate-all.patch
-      ./qemu-patches/syscall.patch
-      ./qemu-patches/qemu-2.3.0-glibc-2.26.patch
-    ];
+  patches = [
+    # patches extracted from afl source
+    ./qemu-patches/elfload.patch
+    ./qemu-patches/cpu-exec.patch
+    ./qemu-patches/syscall.patch
```

Comment:
Can these patches be downloaded instead of living in the nixpkgs source tree?

Reply:
Unlikely - I've hand-modified at least one of them. See the README as to how/why this is necessary.

Reply:
Can you use `fetchpatch` for the unmodified ones?

Reply:
It's a bit odder than that actually - the patches actually come from the afl source tree itself, so really ideally there would be a neat way to fish them out of there...? Question is how to do it without introducing some kind of weird recursive dependency.

Reply:
Either you write a `postPatch` hook that calls `patch` or add something like something like: "${src}/qemu-patches/patch-name.patch" to the patches list. Both should not have any problems with recursion.

Reply:
It's not the qemu source the patches live in, but the afl source. _This_ qemu is a dependency of afl, and to ensure the correct pairing of versions, I'll have to take that afl package (or at least its src file) as an argument to this function to be able to access it, hence my worry about recursion. It should be possible but will require some jiggery pokery.

Reply:
Since you are not depending on the build output but the source, this should not be a problem either. Using "${avl.src}/qemu-patches/patch-name.patch" should work too.

Reply:
Ahok I'll try that